### PR TITLE
Added instance method __len__ for numpy

### PR DESCRIPTION
**Pull request by ahmedelgamal0**
Close https://github.com/unifyai/ivy/issues/9948

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 1]
```diff
@@ -2437,3 +2437,36 @@ def test_numpy_instance_abs__(
         frontend=frontend,
         frontend_method_data=frontend_method_data,
     )
+
+ 
+# __len__
+@given(
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2443]: Use @handle_frontend_method and include the additional parameters like class_tree, init_tree, method_name

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -2437,3 +2437,36 @@ def test_numpy_instance_abs__(
...
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=1,
+        max_num_dims=5,
+    ),
+)
+def test_numpy_instance_len__(
+    dtype_and_x,
+    as_variable,
+    native_array
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2453]: Inlude more flags like num_positional_args_method, frontend_method_data, and frontend.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 3]
```diff
@@ -2437,3 +2437,36 @@ def test_numpy_instance_abs__(
...
+            "data": x[0],
+        },
+        all_as_kwargs_np_method={},
+        as_variable_flags_method=[],
+        num_positional_args_method=0,
+        native_array_flags_method=[],
+        frontend="numpy",
+        class_name="ndarray",
+        method_name="__len__",
+    )
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2472]: Most of these are either wrongly named or hard coded when flags handled by the test should be passed.

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution! The function itself is done correctly but the tests needs some work, I added some comments but please check some other tests in the file and follow their example.

**Comment by fspyridakos** [comment]
Also please open an issue for this in this ToDo: https://github.com/unifyai/ivy/issues/3607 and link it in this PR as stated in our contribution guide.

**Comment by ahmedelgamal0** [comment]
Close #9948

**Comment by fspyridakos** [comment]
Made a small change to get the test passing. Looks good! Many thanks for your contribution!

**Pull request merged**